instrumentor/controllers/startlangdetection/source_controller.go

```diff
@@ -30,54 +31,100 @@ func (r *SourceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	obj := workload.ClientObjectFromWorkloadKind(source.Spec.Workload.Kind)
-	err = r.Client.Get(ctx, types.NamespacedName{Name: source.Spec.Workload.Name, Namespace: source.Spec.Workload.Namespace}, obj)
-	if err != nil {
-		// TODO: Deleted objects should be filtered in the event filter
-		return ctrl.Result{}, err
-	}
-	instConfigName := workload.CalculateWorkloadRuntimeObjectName(source.Spec.Workload.Name, source.Spec.Workload.Kind)
-
 	if source.DeletionTimestamp.IsZero() {
-		if !controllerutil.ContainsFinalizer(source, consts.SourceFinalizer) {
-			controllerutil.AddFinalizer(source, consts.SourceFinalizer)
-			// Removed by deleteinstrumentedapplication controller
+		if !controllerutil.ContainsFinalizer(source, consts.InstrumentedApplicationFinalizer) {
 			controllerutil.AddFinalizer(source, consts.InstrumentedApplicationFinalizer)
+		}
+		if source.Labels == nil {
+			source.Labels = make(map[string]string)
+		}
 
-			if source.Labels == nil {
-				source.Labels = make(map[string]string)
-			}
-			source.Labels[consts.WorkloadNameLabel] = source.Spec.Workload.Name
-			source.Labels[consts.WorkloadNamespaceLabel] = source.Spec.Workload.Namespace
-			source.Labels[consts.WorkloadKindLabel] = string(source.Spec.Workload.Kind)
+		source.Labels[consts.WorkloadNameLabel] = source.Spec.Workload.Name
+		source.Labels[consts.WorkloadNamespaceLabel] = source.Spec.Workload.Namespace
+		source.Labels[consts.WorkloadKindLabel] = string(source.Spec.Workload.Kind)
 
-			if err := r.Update(ctx, source); err != nil {
-				return k8sutils.K8SUpdateErrorHandler(err)
-			}
+		if err := r.Update(ctx, source); err != nil {
+			return k8sutils.K8SUpdateErrorHandler(err)
+		}
 
-			err = requestOdigletsToCalculateRuntimeDetails(ctx, r.Client, instConfigName, req.Namespace, obj, r.Scheme)
+		// pre-process existing Sources for specific workloads so we don't have to make a bunch of API calls
+		// This is used to check if a workload already has an explicit Source, so we don't overwrite its InstrumentationConfig
+		sourceList := v1alpha1.SourceList{}
+		err := r.Client.List(ctx, &sourceList, client.InNamespace(source.Spec.Workload.Name))
```

Comment:
perhaps I got it wrong or something changed, but isn't `Source` resource resides in the `odigos-system` namespace? 
The `client.InNamespace` seems irrelevant here (do we need to us the `consts.WorkloadNamespaceLabel` label instead?)